pkg/operator/controller/ingress/status_test.go

```diff
@@ -2749,19 +2750,121 @@ func Test_MergeConditions(t *testing.T) {
+			}},
+			expected: []operatorv1.OperatorCondition{{
+				Type:   "CanaryChecksSucceeding",
+				Status: "False",
+				// A change to the condition's message should not change the lastTransitionTime.
+				LastTransitionTime: metav1.NewTime(start),
+				Reason:             "CanaryChecksRepetitiveFailures",
+				// Use the updated message.
+				Message: "Canary route checks for the default ingress controller are failing. Last 1 error messages:\nerror sending canary HTTP Request: Timeout: Get \"https://canary-openshift-ingress-canary.apps.example.local\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) (x14 over 15m10s)",
+			}},
+		},
+		"an update to message, reason, and status in CanaryChecksSucceeding should change lastTransitionTime, message, reason, and status": {
+			conditions: []operatorv1.OperatorCondition{{
+				Type:               "CanaryChecksSucceeding",
+				Status:             "True",
+				LastTransitionTime: metav1.NewTime(start),
+				Reason:             "CanaryChecksSucceeding",
+				Message:            "Canary route checks for the default ingress controller are successful",
+			}},
+			updates: []operatorv1.OperatorCondition{{
+				Type:               "CanaryChecksSucceeding",
+				Status:             "False",
+				LastTransitionTime: metav1.NewTime(middle),
+				Reason:             "CanaryChecksRepetitiveFailures",
+				Message:            "Canary route checks for the default ingress controller are failing. Last 1 error messages:\nerror sending canary HTTP Request: Timeout: Get \"https://canary-openshift-ingress-canary.apps.example.local\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) (x5 over 4m40s)",
+			}},
+			expected: []operatorv1.OperatorCondition{{
+				Type:   "CanaryChecksSucceeding",
+				Status: "False",
+				// Update lastTransitionTime if status changed.
+				LastTransitionTime: metav1.NewTime(later),
+				Reason:             "CanaryChecksRepetitiveFailures",
+				Message:            "Canary route checks for the default ingress controller are failing. Last 1 error messages:\nerror sending canary HTTP Request: Timeout: Get \"https://canary-openshift-ingress-canary.apps.example.local\": context deadline exceeded (Client.Timeout exceeded while awaiting headers) (x5 over 4m40s)",
+			}},
+		},
+		"an update to only the status in CanaryChecksSucceeding should change lastTransitionTime and status": {
+			conditions: []operatorv1.OperatorCondition{{
+				Type:               "CanaryChecksSucceeding",
+				Status:             "Unknown",
+				LastTransitionTime: metav1.NewTime(start),
+				Reason:             "CanaryChecksSucceeding",
+				Message:            "Canary route checks for the default ingress controller are successful",
+			}},
```

Comment:
For `Status: "Unknown"`, you probably want `Reason: "CanaryRouteNotAdmitted"` or `Reason: "CanaryRouteDoesNotExist"`, with the corresponding `Message` text.  It doesn't _really_ matter (the specific text of the reason and message don't matter for the test), but it does look odd to have `Status: "Unknown"` with `Reason: "CanaryChecksSucceeding"` and `Message: "Canary route checks for the default ingress controller are successful"`.

Ah, on further thought, if the intention is to have a test case that verifies that `MergeConditions` behaves correctly when updating `Status` without changing `Reason` or `Message`, then maybe it would be clearer to use some generic status, reason, and message text for the test case.  

These are minor details though; feel free to ignore this suggestion.  

(Sorry if you see an extra notification for this comment; I initially added the comment on `https://github.com/openshift/cluster-ingress-operator/commit/523461a512528f13675de1204e3d6ad381974dbd`, but that doesn't show up when viewing the PR, so I'm adding the same comment again on `https://github.com/openshift/cluster-ingress-operator/pull/1125/commits/523461a512528f13675de1204e3d6ad381974dbd`.)

Reply:
Yes, the intention was to have a test test case that verifies that `MergeConditions` behaves correctly when updating `Status` without changing `Reason` or `Message`.  It didn't look that weird to me, so if this is the only thing you'd want to see changed, I hope you don't mind if I keep it.